src/TaxonomyTerm.php

```diff
@@ -99,9 +105,41 @@ public function getCMSFields()
                         )
                 );
             }
-        }
+            $fields->addFieldToTab('Root.Children', $childrenGrid);
+        });
+
+        return parent::getCMSFields();
+    }
+
+    public function scaffoldFormFieldForHasMany(
+        string $relationName,
+        ?string $fieldTitle,
+        DataObject $ownerRecord,
+        bool &$includeInOwnTab
+    ): FormField {
+        $includeInOwnTab = false;
+        return $this->scaffoldFormFieldForManyRelation($relationName, $fieldTitle);
+    }
 
-        return $fields;
+    public function scaffoldFormFieldForManyMany(
+        string $relationName,
+        ?string $fieldTitle,
+        DataObject $ownerRecord,
+        bool &$includeInOwnTab
+    ): FormField {
+        $includeInOwnTab = false;
+        return $this->scaffoldFormFieldForManyRelation($relationName, $fieldTitle);
+    }
+
+    private function scaffoldFormFieldForManyRelation(string $relationName, ?string $fieldTitle): FormField
+    {
+        $list = static::get();
+        $field = SearchableMultiDropdownField::create($relationName, $fieldTitle, $list, labelField: 'Name');
+        // Use the same lazyload threshold has_one relations use
+        $threshold = DBForeignKey::config()->get('dropdown_field_threshold');
+        $overThreshold = $list->count() > $threshold;
+        $field->setIsLazyLoaded($overThreshold)->setLazyLoadLimit($threshold);
+        return $field;
     }
```

Comment:
Other than the `Children` relation within a `TaxonomyTerm` itself (handled in `getCMSFields()` above), you'll almost always want to be using a field like this for taxonomies.